jupyterhub/spawners.py

```diff
@@ -299,6 +299,9 @@ def get_pod_manifest(self):
         # set the image pull policy
         self.image_pull_policy = "Always"
 
+        # Disable service links to prevent a lot of environment variables appearing in user environment
```

Comment:
Maybe a bit clearer to our future selves about the reason for this

```suggestion
        # Prevent kubernetes service links from appearing in user environment
```